test/cpp/jit/test_misc.cpp

```diff
@@ -2399,6 +2399,22 @@ TEST(FuturesTest, Basic) {
   ASSERT_EQ(sat2, 1);
 }
 
+// Sparse CUDA tensor test
+TEST(FutureTest, SparseTensor) {
+  // Skip test if CUDA is not available.
+  bool has_cuda = at::globalContext().hasCUDA();
+  if (!has_cuda) {
+    LOG(INFO) << "CUDA not available, skipping test";
+  }
+  auto f = c10::make_intrusive<Future>(TensorType::get());
+  at::TensorOptions opts = at::TensorOptions().device(at::DeviceType::CUDA);
+  auto sparse_tensor = at::sparse_coo_tensor(torch::ones(10), torch::ones(10), opts);
```

Comment:
The following should work for creating a sparse tensor:

`at::sparse_coo_tensor(torch::ones(1, 10), torch::ones(10, 10), opts);`

Reply:
Would also recommend creating a dense tensor and calling `.to_sparse()` as that will give the indices a different representation

Reply:
thanks will try!